multiview/embed/mvmds.py

```diff
@@ -0,0 +1,240 @@
+            for j in np.arange(self.num_iter):
+                s2 = np.zeros((p, p))
+
+                for yy in np.arange(views):
+                    d2 = n_num[yy] * np.sum(np.array([n] * views))
+
+                    #Dividing by .0001 is to prevent divide by 0 error
+                    if d[:, yy] == 0:
+                        s2 = s2 + (d2 * Xs[yy] / .0001)
+
+                    else:
+                        s2 = s2 + (d2 * Xs[yy] / d[:, yy])
+
+                w = np.dot(s2, q)
+
+                w = np.dot(pi, w)
+
+                q = w / np.sqrt(np.dot(w.T, w))
+
+                for yy in np.arange(views):
+
+                    d[:, yy] = np.dot(np.dot(q.T, Xs[yy]), q)
+
+            #creates next component
+            components[:, i] = q[:, 0]
+            pi = pi - np.dot(q, q.T)
+
+        return(components)
+
+    def fit(self, Xs):
+
+        """
+        Calculates dimensionally reduced components by inputting the Euclidean
+        distances of each view, double centering them, and using the _cpc function
+        to find common components between views. Works similarly to traditional,
+        single-view Multidimensional Scaling.
+
+        Parameters
+        ----------
+
+        Xs: list of array-likes or numpy.ndarray
+                - Xs length: n_views
+                - Xs[i] shape: (n_samples, n_features_i)
+
+        Attributes
+        ----------
+
+        components: A k-dimensional projection of shape [n,k]
+
+        """
+
+        if (self.n_components) > len(Xs[0]):
+            self.n_components = len(Xs[0])
+            warnings.warn('The number of components you have requested is '
+                          + 'greater than the number of samples in the '
+                          + 'dataset. ' + str(self.n_components)
+                          + ' components were computed instead.')
+
+        if (self.num_iter) <= 0:
+            raise ValueError('The number of iterations must be greater than 0')
+        
+        if (self.n_components) <= 0:
+            raise ValueError('The number of components must be greater than 0 '
+                             + 'and less than the number of features')
+
+        for i in Xs:
+            for j in Xs:
+                if i.shape[0] != j.shape[0]:
+                    raise ValueError('Input matrices must have the same '
+                                     + 'number of samples')
+
+        Xs = check_Xs(Xs, multiview=True)
+
+        mat = np.ones(shape=(len(Xs), len(Xs[0]), len(Xs[0])))
+
+        #Double centering ea h view as in single-view MDS
+        for i in np.arange(len(Xs)):
+            view = euclidean_distances(Xs[i])
+            view_squared = np.power(np.array(view), 2)
+
+            J = np.eye(len(view)) - (1/len(view))*np.ones(view.shape)
+            B = -(1/2) * np.matmul(np.matmul(J, view_squared), J)
+            mat[i] = B
+
+        self.components = self._cpc(mat)
+
+    def transform(self, Xs):
```

Comment:
is this for API compliance somehow?

Reply:
I don't know how I feel about this when the transform doesn't make sense for out of sample data, people might get confused and try to use this.

Reply:
Had talked to Ronan about this. I think we decided to do this to keep with the general embedding format. Couldn't split up the function as such. I can keep this as an issue and try to revisit it. I think this relates to making the larger _commonpcs into smaller pieces. This may be a bit of a process because I wrote that function straight from the pseudocode outlined in the paper.